libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -237,8 +237,10 @@ bool SyntaxChecker::visit(FunctionDefinition const& _function)
 		if (v050)
 			m_errorReporter.syntaxError(_function.location(), "Functions without implementation cannot have modifiers.");
 		else
-			m_errorReporter.warning( _function.location(), "Modifiers of functions without implementation are ignored." );
+			m_errorReporter.warning(_function.location(), "Modifiers of functions without implementation are ignored." );
 	}
+	if (_function.name() == "constructor")
+		m_errorReporter.warning(_function.location(), "This function is named \"constructor\" but is not the constructor of the contract.");
```

Comment:
I think we may want a better text here. Like the one we give when using the old-style constructor.